include/boost/thread/pthread/condition_variable.hpp

```diff
@@ -212,8 +234,7 @@ namespace boost
         template<typename lock_type>
         bool timed_wait(lock_type& m,boost::system_time const& abs_time)
         {
-            struct timespec const timeout=detail::to_timespec(abs_time);
-            return do_wait_until(m, timeout);
+            return do_wait_until(m,detail::timespec_to_internal_clock(abs_time));
```

Comment:
I don't agree here. system_time and internal_clock are not convertible

Reply:
See my comments in detail/thread.hpp.